Loop/Managers/DeviceDataManager.swift

```diff
@@ -871,6 +851,36 @@ class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverDelegat
 
     let doseStore: DoseStore
 
+    // MARK: DoseStoreDelegate
+
+    func doseStore(doseStore: DoseStore, hasEventsNeedingUpload pumpEvents: [(objectID: NSManagedObjectID, date: NSDate, dose: DoseEntry?, raw: NSData?)], withCompletion completionHandler: (uploadedObjects: [NSManagedObjectID]) -> Void) {
+
+        guard let uploader = remoteDataManager.nightscoutUploader, device = rileyLinkManager.firstConnectedDevice, pumpModel = pumpState?.pumpModel else {
```

Comment:
Do you regularly use multiple RileyLinks at once? I'm trying to understand what questions you could answer by knowing which one a particular treatment came from. 

On the other hand, I can easily think of questions that would require knowing which pump a treatment came from. Even without the support for easily switching pumps in one app. Shortly, when insulin data is retrieved from history, switching pumps will be totally feasible.